src/dom.ts

```diff
@@ -249,74 +248,87 @@ export class ElementHandle<T extends Node = Node> extends js.JSHandle<T> {
     return result;
   }
 
-  private async _offsetPoint(offset: types.Point): Promise<types.Point | 'invisible'> {
+  private async _offsetPoint(offset: types.Point): Promise<types.Point | 'notvisible'> {
     const [box, border] = await Promise.all([
       this.boundingBox(),
       this._evaluateInUtility(([injected, node]) => injected.getElementBorderWidth(node), {}).catch(e => {}),
     ]);
     if (!box || !border)
-      return 'invisible';
+      return 'notvisible';
     // Make point relative to the padding box to align with offsetX/offsetY.
     return {
       x: box.x + border.left + offset.x,
       y: box.y + border.top + offset.y,
     };
   }
 
-  async _retryPointerAction(progress: Progress, action: (point: types.Point) => Promise<void>, options: PointerActionOptions & types.PointerActionWaitOptions & types.NavigatingActionWaitOptions): Promise<void> {
+  async _retryPointerAction(progress: Progress, action: (point: types.Point) => Promise<void>, options: PointerActionOptions & types.PointerActionWaitOptions & types.NavigatingActionWaitOptions): Promise<'notconnected' | 'done'> {
     let first = true;
     while (progress.isRunning()) {
       progress.log(apiLog, `${first ? 'attempting' : 'retrying'} ${progress.apiName} action`);
       const result = await this._performPointerAction(progress, action, options);
-      if (result === 'done')
-        return;
       first = false;
+      if (result === 'notvisible') {
+        if (options.force)
+          throw new Error('Element is not visible');
+        progress.log(apiLog, '  element is not visible');
```

Comment:
element not visible, retrying...

Reply:
There will be `retrying page.click action` right after this one, from the line 268.